### PR TITLE
TST: Parameterize tests

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by jreback — APPROVED** [review]

File: pandas/tests/indexes/numeric/test_indexing.py [hunk 1]
```diff
@@ -13,17 +13,28 @@ def index_large():
...
-        for method, loc in [("pad", 1), ("backfill", 2), ("nearest", 1)]:
-            assert idx.get_loc(1.1, method) == loc
-            assert idx.get_loc(1.1, method, tolerance=0.9) == loc
+        assert idx.get_loc(1, method=method) == 1
+        assert idx.get_loc(1, method=method, tolerance=0) == 1
 
+    def test_get_loc_float64_method_none(self):
+        idx = Float64Index([0.0, 1.0, 2.0])
+        method = None
+        assert idx.get_loc(1, method=method) == 1
```
Comment (jbrockmendel) [inline pandas/tests/indexes/numeric/test_indexing.py:26]: to match how this looks in other files, lets remove "method_" from the test name, and not pass method kwarg

File: pandas/tests/indexes/numeric/test_indexing.py [hunk 2]
```diff
@@ -13,17 +13,28 @@ def index_large():
...
+
+    @pytest.mark.parametrize(
+        "method, loc", [("pad", 1), ("backfill", 2), ("nearest", 1)]
```
Comment (jbrockmendel) [inline pandas/tests/indexes/numeric/test_indexing.py:29]: for reasons beyond my ken, the pattern is to not put a space after comma

File: pandas/tests/indexes/numeric/test_indexing.py [hunk 3]
```diff
@@ -13,17 +13,28 @@ def index_large():
...
+    )
+    def test_get_loc_float64_method_loc(self, method, loc):
```
Comment (jbrockmendel) [inline pandas/tests/indexes/numeric/test_indexing.py:31]: maybe combine this with test_get_loc_float64_method above?

File: pandas/tests/indexes/numeric/test_indexing.py [hunk 4]
```diff
@@ -82,21 +93,23 @@ def test_get_loc_missing_nan(self):
 
 
 class TestGetIndexer:
-    def test_get_indexer_float64(self):
+    @pytest.mark.parametrize(
+        "method, arr",
+        [("pad", [-1, 0, 1]), ("backfill", [0, 1, 2]), ("nearest", [0, 1, 1])],
+    )
+    def test_get_indexer_float64(self, method, arr):
```
Comment (jbrockmendel) [inline pandas/tests/indexes/numeric/test_indexing.py:100]: "arr" -> "expected"

File: pandas/tests/indexes/numeric/test_indexing.py [hunk 5]
```diff
@@ -82,21 +93,23 @@ def test_get_loc_missing_nan(self):
...
-            idx.get_indexer(idx), np.array([0, 1, 2], dtype=np.intp)
-        )
 
         target = [-0.1, 0.5, 1.1]
         tm.assert_numpy_array_equal(
-            idx.get_indexer(target, "pad"), np.array([-1, 0, 1], dtype=np.intp)
-        )
-        tm.assert_numpy_array_equal(
-            idx.get_indexer(target, "backfill"), np.array([0, 1, 2], dtype=np.intp)
+            idx.get_indexer(target, method=method), np.array(arr, dtype=np.intp)
```
Comment (jbrockmendel) [inline pandas/tests/indexes/numeric/test_indexing.py:105]: separate into three lines 

result = ...
expected = ...
tm.assert_ ...

File: pandas/tests/indexes/numeric/test_indexing.py [hunk 6]
```diff
@@ -228,10 +247,14 @@ def test_take_fill_value_ints(self, klass):
 
 
 class TestContains:
-    def test_contains_float64_nans(self):
-        index = Float64Index([1.0, 2.0, np.nan])
-        assert np.nan in index
-
-    def test_contains_float64_not_nans(self):
+    @pytest.mark.parametrize(
+        "obj",
```
Comment (jbrockmendel) [inline pandas/tests/indexes/numeric/test_indexing.py:251]: obj -> value

**Comment by jbrockmendel** [comment]
a bunch of nitpicks, but this is substantively excellent.  we have tons of legacy tests that need to be split up like this

File: pandas/tests/indexes/numeric/test_indexing.py [hunk 7]
```diff
@@ -13,17 +13,28 @@ def index_large():
 
 
 class TestGetLoc:
-    def test_get_loc_float64(self):
+    @pytest.mark.parametrize("method", ["pad", "backfill", "nearest"])
```
Comment (jreback) [inline pandas/tests/indexes/numeric/test_indexing.py:16]: can create a fixture for this (followon ok)

File: pandas/tests/indexes/numeric/test_indexing.py [hunk 8]
```diff
@@ -82,21 +93,23 @@ def test_get_loc_missing_nan(self):
...
-            idx.get_indexer(target, "backfill"), np.array([0, 1, 2], dtype=np.intp)
+            idx.get_indexer(target, method=method), np.array(arr, dtype=np.intp)
         )
+
+    def test_get_indexer_float64_default_method(self):
+        idx = Float64Index([0.0, 1.0, 2.0])
+
         tm.assert_numpy_array_equal(
-            idx.get_indexer(target, "nearest"), np.array([0, 1, 1], dtype=np.intp)
+            idx.get_indexer(idx), np.array([0, 1, 2], dtype=np.intp)
```
Comment (jreback) [inline pandas/tests/indexes/numeric/test_indexing.py:112]: same here, use result, expected

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexes/numeric/test_indexing.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/indexes/numeric/test_indexing.py:26]: @MomIsBestFriend can you address comments; this is generally a nice refactor

**Comment by WillAyd** [comment]
@MomIsBestFriend can you merge master?

**Comment by jreback** [comment]
@MomIsBestFriend this looks fine, can you merge master and a few comments

**Comment by jreback** [comment]
closing as stale

**Pull request closed without merging**